ethapi/api_test.go

```diff
@@ -52,6 +56,61 @@ func TestGetBlockReceipts(t *testing.T) {
 	}
 }
 
+func TestAPI_GetAccount(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	addr := common.Address{1}
+	codeHash := common.Hash{2}
+	storageRoot := cc.Hash{3}
+	balance := uint256.NewInt(4)
+	nonce := uint64(5)
+	headerRoot := common.Hash{123}
+
+	mockBackend := NewMockBackend(ctrl)
+	mockState := state.NewMockStateDB(ctrl)
+	mockProof := witness.NewMockProof(ctrl)
+	mockHeader := &evmcore.EvmHeader{Root: headerRoot}
+
+	blkNr := rpc.BlockNumberOrHashWithNumber(rpc.LatestBlockNumber)
+
+	mockBackend.EXPECT().StateAndHeaderByNumberOrHash(gomock.Any(), blkNr).Return(mockState, mockHeader, nil)
+	mockState.EXPECT().GetProof(addr, nil).Return(mockProof, nil)
+	mockProof.EXPECT().GetAccountElements(cc.Hash(headerRoot), cc.Address(addr)).Return(nil, storageRoot, true)
+	mockState.EXPECT().GetCodeHash(addr).Return(codeHash)
+	mockState.EXPECT().GetBalance(addr).Return(balance)
+	mockState.EXPECT().GetNonce(addr).Return(nonce)
+	mockState.EXPECT().Error().Return(nil)
+	mockState.EXPECT().Release()
+
+	api := NewPublicBlockChainAPI(mockBackend)
+
+	account, err := api.GetAccount(context.Background(), addr, blkNr)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	if codeHash.Cmp(account.CodeHash) != 0 {
```

Comment:
I think hashes can be compared directly, no need to use `Cmp`.

Also I prefer using` if got, want := ....; got != want {  t.Errorf("unexpected .. , got: %s want %s", got, want) }`
It is IMHO much more readable, and reduces possible errors mixing up variables.

Reply:
btw: why below it uses the framework github.com/stretchr/testify/require
and here it is not used?

Reply:
Changed to use `require.Equal`